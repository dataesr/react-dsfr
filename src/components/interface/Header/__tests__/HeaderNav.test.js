import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  HeaderNav,
} from '../index';
import HeaderContext from '../headerContext';

Enzyme.configure({ adapter: new Adapter() });

describe('<HeaderNav />', () => {
  let wrapper;
  const context = {
    isMobile: true,
    onOpenNav: jest.fn(),
  };

  beforeEach(() => {
    wrapper = (ui, { providerProps, ...renderOptions }) => render(
      <HeaderContext.Provider value={context}>
        {ui}
      </HeaderContext.Provider>,
      renderOptions,
    );
  });
  it('renders correctly', () => {
    const component = renderer
      .create(
        <HeaderNav>
          <li>
            nav #1
          </li>
        </HeaderNav>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should have fr-link--close class when isOpenNav is true', () => {
    wrapper(<HeaderNav><li>nav #1</li></HeaderNav>, { ...context, isOpenNav: true });
    expect(screen.getByRole('button')).toHaveClass('fr-link--close');
  });

  it('should call onOpenNav', () => {
    wrapper(<HeaderNav closeButtonLabel="close"><li>nav #1</li></HeaderNav>, { context });
    userEvent.click(screen.getByText(/close/i));
    expect(context.onOpenNav).toHaveBeenCalled();
  });
});
