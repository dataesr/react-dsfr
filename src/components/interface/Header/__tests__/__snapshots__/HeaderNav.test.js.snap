// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderNav /> renders correctly 1`] = `
<div
  className="fr-header__menu fr-modal"
>
  <div
    className="fr-container"
  >
    <button
      aria-label="fermer la navigation"
      className="fr-link--close fr-link"
      title="Fermer"
      type="button"
    >
      Fermer
    </button>
    <nav
      aria-label="Menu principal"
      className="fr-nav"
      role="navigation"
    >
      <ul
        className="fr-nav__list"
      >
        <li>
          nav #1
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
